sam/models/base_model.py

```diff
@@ -265,65 +238,22 @@ def preprocess_fit(
-
-        assert_contains_nans(
-            X_transformed, "Data cannot contain nans. Imputation not supported for now"
-        )
+        X_trans, y_trans = self.preprocess(X, y, train=True)
+        X_trans, y_trans = remove_until_first_value(X_trans, y_trans)
+        X_trans, y_trans = remove_target_nan(X_trans, y_trans)
+        assert_contains_nans(X_trans, "Data cannot contain nans. Imputation not supported for now")
 
         # Apply transformations to validation data if provided:
         if validation_data is not None:
             X_val, y_val = validation_data
-
             self.validate_data(X_val)
-
-            if self.y_scaler is not None:
-                y_val = pd.Series(
-                    self.y_scaler.transform(y_val.values.reshape(-1, 1)).ravel(),
-                    index=y_val.index,
-                    name=y_val.name,
-                )
-
-            # This does not affect training: it only calls transform, not fit
-            X_val_transformed = self.preprocess_predict(X_val, y_val)
-            X_val_transformed = pd.DataFrame(
-                X_val_transformed, columns=self.get_feature_names(), index=X_val.index
-            )
-            if self.predict_ahead != [0]:
-                y_val_transformed = make_shifted_target(
-                    y_val, self.use_diff_of_y, self.predict_ahead
-                )
-            else:
-                # Dataframe with 1 column. Will use y's index and name
-                y_val_transformed = pd.DataFrame(y_val.copy()).astype(float)
-
-            # Remove the first n rows because they are nan anyway because of rolling features
-            first_complete_index = X_transformed.dropna(axis=0, how="all").index[0]
-            X_transformed = X_transformed.loc[first_complete_index:]
-            y_transformed = y_transformed.loc[first_complete_index:]
-            # The lines below are only to deal with nans in the validation set
-            # These should eventually be replaced by Arjans/Fennos function for removing nan rows
-            # So that this code will be much more readable
-            targetnanrows = y_val_transformed.isna().any(axis=1)
-
-            # Filter rows where the target is unknown
-            X_val_transformed = X_val_transformed.loc[~targetnanrows]
-            y_val_transformed = y_val_transformed.loc[~targetnanrows]
-            # Until here
-            validation_data = (X_val_transformed, y_val_transformed)
+            X_val_trans, y_val_trans = self.preprocess(X_val, y_val, train=False)
+            X_val_trans, y_val_trans = remove_until_first_value(X_val_trans, y_val_trans)
+            X_val_trans, y_val_trans = remove_target_nan(X_val_trans, y_val_trans)
         else:
-            X_val_transformed, y_val_transformed = (None, None)
+            X_val_trans, y_val_trans = (None, None)
 
-        return X_transformed, y_transformed, X_val_transformed, y_val_transformed
+        return X_trans, y_trans, X_val_trans, y_val_trans
```

Comment:
Why did you shorten transformed to trans?

Reply:
It's called `X_transformed` and `y_transformed` in many other functions. I prefer more explicit variable naming, even if it's a bit longer.

Reply:
Changed it back, and found a different way to make it more readable 😛